simulator/src/main/java/com/hedera/block/simulator/BlockStreamSimulatorApp.java

```diff
@@ -60,6 +66,67 @@ public BlockStreamSimulatorApp(
         delayBetweenBlockItems = blockStreamConfig.delayBetweenBlockItems();
     }
 
+    /**
+     * Constructs a new {@code BlockStreamSimulatorApp} with the specified configuration.
+     *
+     * @param configuration the configuration to use for the simulator app
+     */
+    public BlockStreamSimulatorApp(Configuration configuration) {
+        this.configuration = configuration;
+        initilizeDependencies();
+        initializeConfig();
+
+        this.delayBetweenBlockItems = blockStreamConfig.delayBetweenBlockItems();
+    }
+
+    /**
+     * Constructs a new {@code BlockStreamSimulatorApp} with the default configuration.
+     *
+     * @throws IOException if an I/O error occurs while loading the default configuration
+     */
+    public BlockStreamSimulatorApp() throws IOException {
+        this.configuration = loadDefaultConfiguration();
+        initilizeDependencies();
+        initializeConfig();
+
+        this.delayBetweenBlockItems = blockStreamConfig.delayBetweenBlockItems();
+    }
+
+    /**
+     * Initializes the configuration by retrieving the {@code BlockStreamConfig} from the current
+     * configuration.
+     */
+    private void initializeConfig() {
+        this.blockStreamConfig = configuration.getConfigData(BlockStreamConfig.class);
+    }
+
+    /**
+     * Initializes the dependencies required by the simulator app using Dagger dependency injection.
+     */
+    private void initilizeDependencies() {
+        BlockStreamSimulatorInjectionComponent DIComponent =
+                DaggerBlockStreamSimulatorInjectionComponent.factory().create(configuration);
+
+        this.blockStreamManager = DIComponent.getBlockStreamManager();
+        this.publishStreamGrpcClient = DIComponent.getPublishStreamGrpcClient();
+    }
+
+    /**
+     * Loads the default configuration for the simulator app.
+     *
+     * @return the default configuration
+     * @throws IOException if an I/O error occurs while loading the configuration
+     */
+    private Configuration loadDefaultConfiguration() throws IOException {
+        ConfigurationBuilder configurationBuilder =
+                ConfigurationBuilder.create()
+                        .withSource(SystemEnvironmentConfigSource.getInstance())
+                        .withSource(SystemPropertiesConfigSource.getInstance())
+                        .withSource(new ClasspathFileConfigSource(Path.of("app.properties")))
```

Comment:
consider to extract the "app.properties" literal into a class of constants so that we can have a global place for referring to these literals which I have no doubt we will see again.